### DISCUSSION TITLE
AddMessages, page oriented loading translations

### DISCUSSION THREAD

**Discussion by bitdom8** [ivanhofer/typesafe-i18n#322] [Ideas]
First of, this is the most brilliant i18n library out there. Thanks

Personally, I don't like to visit root i18n folger each time. Want my translations on each page. Also Don't want to load all TR translations but just what's necessary for the page. Can we do this addMessages alike function from sveltei18n?

```
if ($locale == 'en') {
			addMessages('en', {
				'Sort by Newest': 'Sort by Newest',
				'Sort by Oldest': 'Sort by Oldest'
			});
		} else {
			addMessages('tr', {
				'Sort by Newest': 'En Yeniye Göre Sırala',
				'Sort by Oldest': 'En Eskiye Göre Sırala'
			});
		}
```

**Comment by ivanhofer** [comment]
Thanks for your kind words!

You should take a look at the [namespace functionality](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/generator#namespaces).
It offers a way to load only the parts that are needed. Let me know if the linked description is clear :)

**Comment by bitdom8** [comment]
That will nicely do it, thanks @ivanhofer !
